processor-api/src/main/java/com/synopsys/integration/alert/processor/api/extract/ProviderMessageExtractionDelegator.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * processor-api
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.processor.api.extract;
+
+import java.util.EnumMap;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.EnumUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.processor.api.detail.ProviderMessageHolder;
+import com.synopsys.integration.alert.processor.api.filter.model.FilterableNotificationWrapper;
+import com.synopsys.integration.blackduck.api.manual.enumeration.NotificationType;
+
+@Component
+public final class ProviderMessageExtractionDelegator {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final EnumMap<NotificationType, ProviderMessageExtractor> notificationTypeToExtractor;
```

Comment:
I'm thinking we may want a class that enforces that a NotificationExtractor and a ProviderMessageExtractor both exist. (Unless the mappings won't be 1 to 1)

Reply:
If we end up using the changes in #1368, then they won't be the same constraints.

Reply:
Also, I'm not sure we could pull it off without separate `NotificationProcessors` for each different notification type.